docs/dev/relay_add_pass.rst

```diff
@@ -0,0 +1,125 @@
+.. _relay-add-pass:
+
+Adding a Compiler Pass to Relay
+===============================
+
```

Comment:
High-level remark about this introductory sections: It should be emphasized that writing passes is the key to adding features to Relay because it is how you traverse the AST and inspect the program.

Reply:
Another good point to emphasize is that Relay's "built in" core language features are implemented using the same pass interface, as a point to show how powerful those are: autodiff and type inference are implemented as passes.